### PR TITLE
compat: /auth: parse server address correctly

### PR DESCRIPTION
**Pull request by vrothberg**
Use `auth.Login` as `podman login` does which parses and normalizes the
input addresses correctly, especially for docker.io.

[NO NEW TESTS NEEDED] as we do not have means to test logging into
docker.io in CI.

Fixes: #17571
Signed-off-by: Valentin Rothberg <vrothberg@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
Fix a bug in the Docker-compatible REST APIs `/auth` endpoint to correctly log into Docker Hub.
```

### CODE REVIEW

File: go.mod
Comment (Luap99) [inline go.mod:17]: is this a downgrade? I ask because github.com/vbauerster/mpb/v7 is added back in.

**Comment by edsantiago** [comment]
WHOA. [Test failures](https://api.cirrus-ci.com/v1/artifact/task/6337919997181952/html/sys-podman-fedora-36-root-host.log.html#t--00208):
```
# podman --events-backend=file events --stream=false --filter type=image --since 2023-02-20T11:00:02-06:00
[fail expected this got that]
```
and
```
#|     FAIL: Image found in events
#| expected: =~ \"Name\":\"quay.io/libpod/testimage:20221018\"
#|   actual:    \{\"ID\":\"f5a99120db6452661930a1db3bf7390eec9b963f5f62c068fa32dc1d550afad3\"\,
\"Name\":\"f5a99120db6452661930a1db3bf7390eec9b963f5f62c068fa32dc1d550afad3\"\,
\"Status\":\"pull\"\,
\"Time\":\"2023-02-20T11:00:13.133325023-06:00\"\,
\"Type\":\"image\"\,\"Attributes\":null\}
```
...look exactly the same as [unexplained but persistent failure in nightly treadmill job](https://api.cirrus-ci.com/v1/artifact/task/6181517337034752/html/sys-podman-fedora-37-root-host.log.html#t--00208)

@containers/podman-maintainers PTAL, this looks serious.

**Comment by vrothberg** [comment]
I probably broke that. Will take a look tomorrow!

**Comment by vrothberg** [comment]
https://github.com/containers/common/pull/1339 will fix it.  Once that got in, I'll update the PR here along with minor tweaks to the system test.

**Comment by vrothberg** [comment]
Breaking out the c/common vendor into https://github.com/containers/podman/pull/17591.

This should buy me some time looking into what breaks because of the /auth changes.

**Comment by vrothberg** [comment]
@Luap99 @edsantiago PTAL

I'm not happy about the error check but it's a smaller problem than the bug.  Will look into separately as changing the error would impact skopeo and buildah as well.

File: pkg/api/handlers/compat/auth.go
Comment (edsantiago) [inline pkg/api/handlers/compat/auth.go:58]: I would be happier with a more distinctive string than "logging into ", but I've at least confirmed that this string does not (currently) appear anywhere except `vendor/github.com/containers/common/pkg/auth/auth.go`

**Review by edsantiago — APPROVED** [review]

File: pkg/api/handlers/compat/auth.go
Comment (vrothberg) [reply pkg/api/handlers/compat/auth.go:58]: Let's hold this PR for https://github.com/containers/common/pull/1340 to get a clean solution.

Comment (Luap99) [inline pkg/api/handlers/compat/auth.go:58]: Can we wrap the error instead? Or alternatively return a new typed error. These string matches break easily and there doesn't seem to be a test for it either.

Comment (Luap99) [reply pkg/api/handlers/compat/auth.go:58]: Ignore this comment: Looks like you are faster than me.

**Comment by vrothberg** [comment]
Updated. Vendored in c/common to keep the error handling.

**Comment by rhatdan** [comment]
LGTM

File: go.mod
Comment (Luap99) [inline go.mod:17]: This is a downgrade.

Comment (vrothberg) [reply go.mod:17]: sigh ... stupid Go.

Updating c/image in c/common does the following:
```
common (main) $ go get github.com/containers/image/v5@10858b2058d8
panic: internal error: can't find reason for requirement on github.com/google/pprof@v0.0.0-20210407192527-94a9f03dee38

goroutine 1 [running]:
cmd/go/internal/modget.(*resolver).updateBuildList.func1({{0xc000703230?, 0xc000179620?}, {0xc000030c30?, 0xc0004a9630?}})
        /usr/lib/golang/src/cmd/go/internal/modget/get.go:1760 +0x114
cmd/go/internal/modget.(*resolver).updateBuildList(0xc000708900, {0xb313b0, 0xc0000280b8}, {0x0, 0x0, 0x0})
        /usr/lib/golang/src/cmd/go/internal/modget/get.go:1765 +0x597
cmd/go/internal/modget.(*resolver).resolveQueries(0xc000708900, {0xb313b0, 0xc0000280b8}, {0xc000014050, 0x1, 0x472485?})
        /usr/lib/golang/src/cmd/go/internal/modget/get.go:1243 +0x1e5
cmd/go/internal/modget.runGet({0xb313b0, 0xc0000280b8}, 0xc00002c510?, {0xc0000241a0, 0x1, 0x1})
        /usr/lib/golang/src/cmd/go/internal/modget/get.go:314 +0x40b
main.invoke(0xe2d000, {0xc000024190, 0x2, 0x2})
        /usr/lib/golang/src/cmd/go/main.go:225 +0x34e
main.main()
        /usr/lib/golang/src/cmd/go/main.go:179 +0x7d1
```

Comment (Luap99) [reply go.mod:17]: what?! how is this even possible

**Comment by mheon** [comment]
Changes LGTM

**Review by Luap99 — COMMENTED** [review]
You need to update the tests, the runtime-spec update changed the version the tests expect: https://github.com/containers/podman/blob/1cfb29d8ed735238998308eca2497ebdecbd71fe/libpod/container_internal_test.go#L181

**Comment by vrothberg** [comment]
> You need to update the tests, the runtime-spec update changed the version the tests expect:
> 
> https://github.com/containers/podman/blob/1cfb29d8ed735238998308eca2497ebdecbd71fe/libpod/container_internal_test.go#L181

Thanks!  Will do.  Had to focus on https://github.com/containers/podman/pull/17615 until now.

**Comment by vrothberg** [comment]
Rebased, should pass now

**Comment by vrothberg** [comment]
@rhatdan @Luap99 merge me please

**Comment by vrothberg** [comment]
or @edsantiago :)

File: pkg/api/handlers/compat/auth.go
Comment (vrothberg) [reply pkg/api/handlers/compat/auth.go:56]: Nice catch!  That part of the comment isn't relevant anymore. In fact, I will remove the entire comment.

**Pull request merged**
